### ISSUE TITLE
CMake Does not install Desktop Files

### ISSUE THREAD

**Issue by diabl0w** [debrouxl/tilp_and_gfm#76]
Hi, I apologize if this is user error, but building using CMake based on the travis.yml script, the install does not seem to create a .desktop file... I am using a prefix of /usr, and I would expect the desktop file to be installed into /usr/share/applications

**Comment by diabl0w** [comment]
Closing this, I removed the extra language part of the CMake and I think that is what is responsible for controlling that part

**Comment by diabl0w** [comment]
sorry for the noise, but reopening, did a clean install of gtk3 branch and no desktop entries to be found, instead the installer puts them here for some reason:
```-- Installing: /usr/share/tilp2/desktop/tilp.appdata.xml
-- Installing: /usr/share/tilp2/desktop/Makefile.am
-- Installing: /usr/share/tilp2/desktop/README.txt
-- Installing: /usr/share/tilp2/desktop/tilp.desktop.in
-- Installing: /usr/share/tilp2/desktop/tilp.xml```

**Comment by adriweb** [comment]
Well idk about desktop files, I just launch `tilp2` from the terminal.

But also. I guess this repo (upstream) hasn't yet gotten updated with the GitHub actions parts replacing the obsolete Travis stuff (although it probably doesn't change anything regarding calling cmake). You can find more recent things here until it gets merged: https://github.com/adriweb/tilp_and_gfm/blob/4a4a0291d31c6e732084b1b5d3fc4cb87957d9cb/.github/workflows/build.linux.workflow.yml#L54

note that this branch that adds GitHub Actions is based on the gtk3 branch (but that's maybe what you want anyway)

**Comment by diabl0w** [comment]
> Well idk about desktop files, I just launch `tilp2` from the terminal.
> 
> But also. I guess this repo (upstream) hasn't yet gotten updated with the GitHub actions parts replacing the obsolete Travis stuff (although it probably doesn't change anything regarding calling cmake). You can find more recent things here until it gets merged: https://github.com/adriweb/tilp_and_gfm/blob/4a4a0291d31c6e732084b1b5d3fc4cb87957d9cb/.github/workflows/build.linux.workflow.yml#L54
> 
> note that this branch that adds GitHub Actions is based on the gtk3 branch (but that's maybe what you want anyway)

of course that is possible to launch from terminal but it is pretty standard for gui apps to have desktop files so that they can be launched from a launcher. and your repo already has the desktop files and also the icon files to properly display them, they just dont get placed properly.

 i know this is issue is low priority, but its also probably an easy fix using an `if Linux` combined with `$PREFIX/share/applications` etc

**Comment by diabl0w** [comment]
> I know, and they're supposed to be installed correctly so maybe there's a target path issue somewhere yes. It's around here: https://github.com/adriweb/tilp_and_gfm/blob/4a4a0291d31c6e732084b1b5d3fc4cb87957d9cb/tilp/trunk/CMakeLists.txt#L125 (But all I was saying is that I wouldn't have known about it anyway, and I guess you're the first to report it)

yes that part of the makefile gets executed correctly, but it expsnds to /usr/share/tilp2 as seen in my stdout log above... whereas those different files belong in multiple different specific directories like the manpage one below it... i tried googling the "proper" way to handle these but to no avail as of yet, until then i just manually copy

**Comment by debrouxl** [comment]
The right way to launch TILP is indeed from a terminal, so as to obtain logging output more easily if need be.
However, this does not mean that the CMake definitions don't need to be fixed, if they are to replace the autotools definitions as the main way to build libti*/gfm/tilp :)

**Comment by adriweb** [comment]
I took a look at it, and there's actually a configure step too to take care of (see what the Makefile.am does), before we can properly install the files to their target location.
